src/overlays/actors/ovl_En_Firefly/z_en_firefly.c

```diff
@@ -21,64 +54,761 @@ const ActorInit En_Firefly_InitVars = {
+}
+
+void EnFirefly_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFirefly* this = THIS;
+
+    Collider_DestroySphere(globalCtx, &this->collider);
+}
+
+void EnFirefly_SpawnIceEffects(EnFirefly* this, GlobalContext* globalCtx) {
+    if (this->unk_18F == 0xA) {
+        this->unk_18F = 0;
+        this->unk_2E8.x = 0.0f;
+        func_800BF7CC(globalCtx, &this->actor, &this->unk_2F8, 3, 2, 0.2f, 0.2f);
+    }
+}
+
+void EnFirefly_Extinguish(EnFirefly* this) {
+    this->currentType = KEESE_NORMAL;
+    this->collider.info.toucher.effect = 0; // Nothing
+    this->auraType = KEESE_AURA_NONE;
+    this->actor.hintId = 0x12; // Keese
+}
+
+void EnFirefly_Ignite(EnFirefly* this) {
+    if (this->actor.params == KEESE_FIRE_FLY) {
+        this->currentType = KEESE_FIRE;
+        this->collider.info.toucher.effect = 1; // Fire
+        this->auraType = KEESE_AURA_FIRE;
+        this->actor.hintId = 0x11; // Fire Keese
+    }
+}
+
+s32 EnFirefly_ReturnToPerch(EnFirefly* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    f32 distFromHome;
+
+    if (this->actor.params != KEESE_NORMAL_PERCH) {
+        return false;
+    }
+
+    if (Actor_XZDistanceToPoint(&player->base, &this->actor.home.pos) > 300.0f) {
+        distFromHome = Actor_DistanceToPoint(&this->actor, &this->actor.home.pos);
+
+        if (distFromHome < 5.0f) {
+            EnFirefly_SetupPerch(this);
+            return true;
+        }
+
+        if (distFromHome * 0.05f < 1.0f) {
+            this->actor.speedXZ *= distFromHome * 0.05f;
+        }
+
+        Math_ScaledStepToS(&this->actor.shape.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x300);
+        Math_ScaledStepToS(&this->actor.shape.rot.x, Actor_PitchToPoint(&this->actor, &this->actor.home.pos) + 0x1554,
+                           0x100);
+
+        return true;
+    }
+
+    return false;
+}
+
+s32 EnFirefly_SeekTorch(EnFirefly* this, GlobalContext* globalCtx) {
+    ObjSyokudai* findTorch;
+    ObjSyokudai* closestTorch;
+    f32 currentDist;
+    f32 currentMinDist;
+    Vec3f flamePos;
+
+    findTorch = (ObjSyokudai*)globalCtx->actorCtx.actorList[ACTORCAT_PROP].first;
+    closestTorch = NULL;
+    currentMinDist = 35000.0f;
+
+    if ((this->actor.params != KEESE_FIRE_FLY) || (this->currentType != KEESE_NORMAL)) {
+        return false;
+    }
+
+    while (findTorch != NULL) {
+        if ((findTorch->actor.id == ACTOR_OBJ_SYOKUDAI) && (findTorch->unk_1DC != 0)) {
+            currentDist = Actor_DistanceBetweenActors(&this->actor, &findTorch->actor);
+            if (currentDist < currentMinDist) {
+                currentMinDist = currentDist;
+                closestTorch = findTorch;
+            }
+        }
+        findTorch = (ObjSyokudai*)findTorch->actor.next;
+    }
+
+    if (closestTorch != NULL) {
+        flamePos.x = closestTorch->actor.world.pos.x;
+        flamePos.y = closestTorch->actor.world.pos.y + 52.0f + 30.0f;
```

Comment:
```suggestion
        flamePos.y = closestTorch->actor.world.pos.y + 82.0f;
```
I'm guessing this won't match because of rodata shenanigans?

Reply:
Yeah, no way would the compiler split that up otherwise. Most likely the numbers will make sense when ObjSyokudai is decomped.